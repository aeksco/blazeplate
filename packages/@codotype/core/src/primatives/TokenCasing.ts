import { TokenCasing } from "../token";
import { buildTokenCasing } from "../util";

// // // //

interface TokenCasingBuilderParams {
    title: string;
}

export class TokenCasingBuilder implements TokenCasing {
    title: string = "";
    snake: string = "";
    camel: string = "";
    pascal: string = "";
    kebab: string = "";

    constructor(params: TokenCasingBuilderParams) {
        const tokenCasing: TokenCasing = buildTokenCasing(params.title);
        this.title = tokenCasing.title;
        this.snake = tokenCasing.snake;
        this.camel = tokenCasing.camel;
        this.pascal = tokenCasing.pascal;
        this.kebab = tokenCasing.kebab;
    }
}
