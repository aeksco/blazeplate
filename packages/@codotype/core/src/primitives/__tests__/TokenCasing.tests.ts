import { TokenCasingBuilder } from "../TokenCasing";

// // // //

describe("Primitives.TokenCasing", () => {
    test("works", () => {
        const tokenCasing = new TokenCasingBuilder({
            title: "User",
        });

        expect(tokenCasing).toMatchSnapshot();
    });
});
