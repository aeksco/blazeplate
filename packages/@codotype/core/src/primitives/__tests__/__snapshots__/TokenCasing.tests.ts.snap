// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Primitives.TokenCasing works 1`] = `
TokenCasingBuilder {
  "camel": "user",
  "kebab": "user",
  "pascal": "User",
  "snake": "user",
  "title": "User",
}
`;
